### ISSUE TITLE
General feedback on Tatsu

### ISSUE THREAD

**Issue by stpaultim** [backdrop-contrib/tatsu#47]
Have you tried Tatsu? Are you using Tatsu?

Please, leave a note hear with your feedback or comments. 
I'd like like to hear if anyone is using it and if not, why not (recognizing that it's still a little rough).

**Comment by keiserjb** [comment]
I'm trying to use it.  I like the bootstrap support and that you modified the header so the menu collapses into the hamburger.

The last couple days I've gotten around to making some custom templates.  After doing this I realized I should probably make a sub-theme.  After doing that and enabling it I lost the ability to change the colors through the user interface.  I must've done something wrong.

**Comment by stpaultim** [comment]
I will reply later tonight. I have ideas.

**Comment by oadaeh** [comment]
> After doing that and enabling it I lost the ability to change the colors through the user interface. I must've done something wrong.

Is that possibly related to https://github.com/backdrop/backdrop-issues/issues/3348 ?

**Comment by stpaultim** [comment]
Yes, making a sub-theme is a good idea. I have not done it yet, because I've been trying to incorporate most of my own changes into the Tatsu itself. But, making a sub-theme is a good idea.

Check this section of the sub-theming instructions for Backdrop CMS and let me know if this helps you. If so, we should including instructions in Tatsu to help users with this.  

> **Copy color module settings if needed**
> As the sections below indicate, the sub-theme inherits most properties of the base theme. The important exceptions are core version and color info. You probably want to copy the core version declaration.
> 
> If your base theme supports the color module and you'd like your sub-theme to support it, you probably also want to copy the color folder from your base theme and add the line from your base theme's info file to your sub-themes info file that looks like
> 
> stylesheets[all][] = css/colors.css
> 
> and then copy the colors.css from your base theme to the css folder of your sub-theme.

EDIT: I had not seen the comment by @oadaeh when I replied. That might also be the problem. Although, technically, Tatsu is a sub-theme of Basis and color module has been working for me in Tatsu. Although, Tatsu is not trying to inherit color module support from Basis. I think we have everything we need directly in Tatsu.

**Comment by stpaultim** [comment]
Here is my quick understanding of how Color Modules works to help you figure out how to resolve your problem. 

1. In the color/color.inc file, the color modules lists all of the color settings from the theme and their default color. 

2. It also creates a list of CSS files to watch and change based upon the settings in the theme. 

3. Anytime a user changes colors using the UI settings for the theme, the color module goes through all the css files listed in color/color.inc, it looks for anything using any of the default colors and changes them as necessary, and it writes NEW css files in the files/ directory. These new CSS files will override the css in your theme. 

NOTE: If you make changes to the colors in your theme, all of the css in the designated files will be copied into the files directory and over ride the your original css files. This includes all of the CSS, even if it is not directly effected by your color changes. 

There is an assumption, that once you start using the color module to over ride colors, you will be editing those specific css files in the theme anymore. 

Sometimes, you may need to simply "resave" the colors on your themes setting page.

**Comment by keiserjb** [comment]
Part of my confusion lies in that I cannot locate the colors.css file.  But after copying over the contents of the tatsu.info file the colors match in my sub-theme.  However under appearance and settings for the sub-theme there's still no way to change the colors.  I can just edit the tatsu settings though and the colors change.

**Comment by stpaultim** [comment]
I think that I need to focus on how to sub-theme Tatsu this weekend. I did a really quick test, creating a super simple sub-theme of Tatsu and the color module works fine for me. But, I am not sure how realistic my simple test is 

Just to be clear, the color module does not automatically create a colors.css file. In your theme there should be a file called color.inc. In the color.inc file is list of css files that you want the color module to watch. Any time you save changes, through the theme settings page your theme will go through all of the css files and make copies of them in your files directory. The color module will change the new css files to reflect your color changes. These new files will over ride the default files in your theme. 

Does this make any sense?

I think it might make sense to put all the color changes into a single color.css file. 

![Tatsu](https://user-images.githubusercontent.com/3144571/68537879-f6abf700-0330-11ea-835c-388264b6ffa4.png)

**Comment by stpaultim** [comment]
@keiserjb - This might help you. I would love any feedback or suggestions:

https://www.triplo.co/blog/color-module-backdrop-cms

**Comment by keiserjb** [comment]
Thanks for spending some time with this.  Your screenshot above where there's a folder for drippy in the color folder in the files directory, I'm not getting one of those for my sub-theme.

**Comment by stpaultim** [comment]
files/color/drippy-406d1787 is the css for the "drippy" sub-theme. This file should be created once you edit the color scheme in the configuration settings for your theme. There is no need for this file unless you actually make changes to the color scheme. This file contains the overrides. 

Have you read this blog post - https://www.triplo.co/blog/color-module-backdrop-cms

I think it should be helpful. If it's not, let me know how it can be improved or what information is missing.

**Comment by keiserjb** [comment]
Bingo, I got it.

**Comment by yorkshire-pudding** [comment]
Is an improvement on Basis.  I was having a few issues until I switched manually to 1.2.  This is an improvement. Because Tatsu uses Basis colour schemes as a base, it inherits the https://github.com/backdrop/backdrop-issues/issues/4415 issue so if you want to use Goldfish or Ice you have to customise the colours to avoid this issue.

**Comment by stpaultim** [comment]
Thanks for the feedback. This might be the inspiration I need to focus on Tatsu more.

**Comment by stpaultim** [comment]
@yorkshire-pudding and I had a meeting today to talk about the future of Tatsu and the possibility of creating a 1.2.x release. 

We identified a few issues we'd like to work on before issuing a 1.2.x release, I'll try to either comment in other issues or create new ones. BUT, I'm committed to working on Tatsu and getting out a new release in the near future. Now would be a great time for anyone to make comments or suggestions. 

PR's welcome

**Comment by izmeez** [comment]
Just testing tatsu-1.x-2.x with manual install. Prefer it to Basis because of how it handles the Logo and menus. It definitely has wrinkles that I shall try to append to the individual issues.

**Comment by stpaultim** [comment]
I've been getting some feedback on Tatsu, but haven't had time to work on it. I will try to find time in the next few weeks to reply to issue comments and maybe fix some of the issues mentioned.
